src/Application.php

```diff
@@ -132,17 +132,23 @@ public function __call($method, $args)
         if (! in_array(strtoupper($method), $this->httpRouteMethods, true)) {
             throw new BadMethodCallException('Unsupported method');
         }
-
-        if (count($args) !== 2) {
+        $numArgs = count($args);
+        if ($numArgs < 2) {
             throw new BadMethodCallException(sprintf(
-                '%s::%s requires exactly 2 arguments; received %d',
+                '%s::%s requires at least 2 arguments; received %d',
                 __CLASS__,
                 $method,
                 count($args)
             ));
         }
 
-        $args[] = [$method];
+        if ($numArgs === 3) {
+            $name    = $args[2];
+            $args[2] = [$method];
+            $args[]  = $name;
```

Comment:
@weierophinney if you use setName() on the returned Route object, you are not changing the name of the route for the Application, because of [this](https://github.com/ezimuel/zend-expressive/blob/fix/40/src/Application.php#L297-L298).